### PR TITLE
Fix Zombie ShouldBurnInDay API (#4268)

### LINKED ISSUE #4268: Zombie#setShouldBurnInDay does not work
### What behaviour is expected:
I expect the zombie, that was flagged with Zombie#setShouldBurnInDay(false) to not burn in daylight.

### What behaviour is observed:
The zombie burns nevertheless.

### Steps/models to reproduce:
* Spawn a zombie with something like this: `Zombie en = world.spawn(location, Zombie.class);`
* Set "shouldBurnInDay" to false like this: ` en.setShouldBurnInDay(false);`
* Still see it burning

### Plugin list:
Only the test plugin, containing only this functionality

### Paper version:
This server is running Paper version git-Paper-174 (MC: 1.16.2) (Implementing API version 1.16.2-R0.1-SNAPSHOT)

### Anything else:
I already looked into the code and saw the obvious reason for the behaviour observed in the EntityZombie.class:
```java
    public boolean shouldBurnInDay() {
        return this.T_();
    }

    protected boolean T_() {
        return true;
    }

    public void setShouldBurnInDay(boolean shouldBurnInDay) {
        this.shouldBurnInDay = shouldBurnInDay;
    }
```

I also checked if this was somehow noted in the JavaDoc through for example deprecating the method, but this is not the case: https://papermc.io/javadocs/paper/1.16/org/bukkit/entity/Zombie.html#setShouldBurnInDay-boolean-

Thanks for looking into this!

### PR DESCRIPTION
**Pull request by scrayos**
Fixes https://github.com/PaperMC/Paper/issues/4268
(Replaces https://github.com/PaperMC/Paper/pull/4270)

### CODE REVIEW

File: Spigot-Server-Patches/0319-Add-more-Zombie-API.patch [hunk 1]
```diff
@@ -29,24 +29,24 @@ index 12f954723ac2f0058081d810ce83aed99532caad..02a845f99bb445fb4aa66a90273469a8
...
 +    public void setShouldBurnInDay(boolean shouldBurnInDay) {
 +        this.shouldBurnInDay = shouldBurnInDay;
-+    }
+     }
 +    // Paper end
-+
+ 
      @Override
      public boolean damageEntity(DamageSource damagesource, float f) {
-         if (!super.damageEntity(damagesource, f)) {
```
Comment (Prof-Bloodstone) [inline Spigot-Server-Patches/0319-Add-more-Zombie-API.patch:49]: You seem to have deleted a line you shouldn't.

File: Spigot-Server-Patches/0319-Add-more-Zombie-API.patch [hunk 2]
```diff
@@ -29,24 +29,24 @@ index 12f954723ac2f0058081d810ce83aed99532caad..02a845f99bb445fb4aa66a90273469a8
...
++    }
++
 +    // Paper start
 +    public void setShouldBurnInDay(boolean shouldBurnInDay) {
 +        this.shouldBurnInDay = shouldBurnInDay;
-+    }
+     }
 +    // Paper end
-+
+
```
Comment (Prof-Bloodstone) [inline Spigot-Server-Patches/0319-Add-more-Zombie-API.patch:47]: I don't really understand why is there so many changes here - it should be just:
```diff
  protected boolean T_() {
-   return true;
+   return this.shouldBurnInDay; // Paper - use api value instead
  }
```

**Review by Prof-Bloodstone — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0319-Add-more-Zombie-API.patch (hunk 2 above)
Comment (scrayos) [reply Spigot-Server-Patches/0319-Add-more-Zombie-API.patch:47]: I only modified the one line. I guess that the other "changed" lines are caused by the usual heuristics that git performs in order to calculate diffs. It changed which brace it used/where it injects the patch content into the file. But I did not change any line except for the return.

**Review by Proximyst — DISMISSED** [review]

File: Spigot-Server-Patches/0319-Add-more-Zombie-API.patch (hunk 1 above)
Comment (scrayos) [reply Spigot-Server-Patches/0319-Add-more-Zombie-API.patch:49]: I just double checked and I definitely did not even touch, let alone delete this line. AFAIK the first sign is the git diff, the second sign (that is missing in the change) would be the patch diff. So lines in patch-commits that only have the first sign should be irrelevant, as they only modified which lines are included in the context of the file in the patch. This probably happened automatically as I only did the following:
```
./paper patch
./paper edit server
<performed the one-line edit>
./paper edit continue
<commited the change>
```

**Review by Octol1ttle — DISMISSED** [review]

**Review by electronicboy — DISMISSED** [review]

**Comment by MiniDigger** [comment]
thanks!

@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Pull request merged**